code/__DEFINES/say.dm

```diff
@@ -102,6 +102,5 @@
 #define INVOCATION_EMOTE "emote"
 #define INVOCATION_WHISPER "whisper"
 
-//Used in visible_message_flags, audible_message_flags and runechat_flags
-#define EMOTE_MESSAGE (1<<0)
-#define RADIO_MESSAGE (1<<1)
+//Used in visible_message_flags, audible_message_flags and message_mods
+#define CHATMESSAGE_EMOTE "emotemessage"
```

Comment:
What's the reason of changing it? Isn't finding in list slower than checking for bit flag? Also, why is radio message removed?

Reply:
Previously when something was heard, the message mods would be taken and checked for radio messages. If the message mod was a radio message then it would pass the radio_message flag in. Instead of doing the conversion, I just directly pass the message mods to the create image proc. Chat message emote is still here because on emotes which have no message mods, it needs something to identify that its an emote and there is no emote mode.
Not sure about the speed difference between finding something in the list, however previously normal messages looked for stuff in the list anyway before converting them to flags.

Reply:
I tried to do it as an assoc list check first, but for some reason that runtimed. Presummable something was being passed in in a non assoc format in message_mods.